### PR TITLE
Client sending name

### PR DESCRIPTION
**Pull request by mprtcz**
For reviewers:
All the information about current changes is in the commit's messages.
The main concern for this pull request would be corectness of mocking tests

### CODE REVIEW

File: src/test/java/com/spanish_inquisition/battleship/client/network/SocketClientTest.java [hunk 1]
```diff
@@ -2,39 +2,63 @@
...
+        final Socket socket = mock(Socket.class);
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(emptyPayload);
+        when(socket.getOutputStream()).thenReturn(byteArrayOutputStream);
+        when(socket.getInputStream()).thenReturn(byteArrayInputStream);
+
+        SocketClient socketClient = new SocketClient();
+        socketClient.socket = socket;
+        socketClient.setUpStreamsOnSocket();
+        assertEquals(byteArrayOutputStream.size(), 0);
```
Comment (agawda) [inline src/test/java/com/spanish_inquisition/battleship/client/network/SocketClientTest.java]: Is this necessary?

File: src/main/java/com/spanish_inquisition/battleship/client/network/SocketClient.java [hunk 2]
```diff
@@ -8,23 +8,41 @@
...
 
+import static com.spanish_inquisition.battleship.common.AppLogger.DEFAULT_LEVEL;
+
 /**
  * @author Michal_Partacz
  */
 public class SocketClient {
     private final static String HOST_NAME = "localhost";
-    private final static int PORT = 6666;
+    final static int PORT = 6666;
```
Comment (agawda) [inline src/main/java/com/spanish_inquisition/battleship/client/network/SocketClient.java]: :smiling_imp:

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 3]
```diff
@@ -1,24 +1,62 @@
...
+    public SocketClient socketClient;
+
     /**
      * This method is run automatically right after the fxml file's loaded
      */
     @FXML
-    public void initialize() {}
+    public void initialize() {
+        try {
+            this.socketClient = SocketClient.createSocketClientWithSocket();
```
Comment (agawda) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: Isn't it possible to inject client socket into this class?

Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: It is, though it is not needed in the Controller class, let's leave the socket stuff in dedicated socket class.

File: src/test/java/com/spanish_inquisition/battleship/client/network/SocketClientTest.java (hunk 1 above)
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/client/network/SocketClientTest.java]: Not really, but better be safe than sorry.

File: src/main/java/com/spanish_inquisition/battleship/client/network/SocketClient.java [hunk 4]
```diff
@@ -8,23 +8,41 @@
...
+
+    SocketClient() {}
 
-    void setUpStreamsAndOpenSocket() throws IOException {
-        logger.log(level, "Setting up streams and socket");
-        socket = new Socket(HOST_NAME, PORT);
+    public void setUpStreamsOnSocket() throws IOException {
+        logger.log(DEFAULT_LEVEL, "Setting up streams and socket");
         input = new BufferedReader(new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8));
         output = new PrintWriter(new OutputStreamWriter(socket.getOutputStream(), StandardCharsets.UTF_8), true);
```
Comment (arturAR) [inline src/main/java/com/spanish_inquisition/battleship/client/network/SocketClient.java]: Encoding 👍

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java [hunk 5]
```diff
@@ -41,12 +38,12 @@ public static void main(String[] args) {
         return clientSockets;
     }
 
-    static ServerSocket createServerSocket() {
+    static ServerSocket createServerSocket(int portNumber) {
         ServerSocket serverSocket = null;
```
Comment (arturAR) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java]: This is dangerous...

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 6]
```diff
@@ -1,24 +1,62 @@
...
+        } catch (IOException e) {
+            logger.log(Level.WARNING, "The client could not connect to the server", e);
+            gameStatusLabel.setText("I could not connect to the server :(");
+        }
+        initializeElementsAfterServerConnection();
+    }
+
+    private void initializeElementsAfterServerConnection() {
+        this.centralVBox.setVisible(true);
```
Comment (arturAR) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: Why always "this"?

File: src/main/java/com/spanish_inquisition/battleship/client/network/SocketClient.java [hunk 7]
```diff
@@ -8,23 +8,41 @@
...
+    public static SocketClient createSocketClientWithSocket() throws IOException {
+        SocketClient socketClient = new SocketClient();
+        socketClient.socket = new Socket(HOST_NAME, PORT);
+        socketClient.setUpStreamsOnSocket();
+        return socketClient;
+    }
+
+    public void sendStringToServer(String string) {
+        try {
+            this.output.println(string);
```
Comment (arturAR) [inline src/main/java/com/spanish_inquisition/battleship/client/network/SocketClient.java]: Maybe use optional class?

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java (hunk 6 above)
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: To not confuse method variables with instance fields;

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 8]
```diff
@@ -1,24 +1,62 @@
...
  * in the file's UI elements.
  */
 public class MainMenuController {
-
     public Button featureButton;
+    public Label gameStatusLabel;
+    public TextField nameTextField;
+    public VBox centralVBox;
+
+    public SocketClient socketClient;
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: access can be private

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java (hunk 3 above)
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: I agree with @mprtcz

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 9]
```diff
@@ -1,24 +1,62 @@
...
+    private void initializeElementsAfterServerConnection() {
+        this.centralVBox.setVisible(true);
+    }
 
+    /**
+     * After the button is clicked the information entered in the textField is sent to the server
+     */
     @FXML
     public void onFeatureButtonClicked() {
-        System.out.println("Clicked!");
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: 👍

File: src/test/java/com/spanish_inquisition/battleship/client/network/SocketClientTest.java [hunk 10]
```diff
@@ -2,39 +2,63 @@
...
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.*;
```
Comment (piotr921) [inline src/test/java/com/spanish_inquisition/battleship/client/network/SocketClientTest.java]: import '*' 🙈

**Pull request merged**